backup_monkey/core.py

```diff
@@ -115,7 +119,22 @@ def snapshot_volumes(self):
                 description_parts.append(volume.attach_data.device)
             description = ' '.join(description_parts)
             log.info('Creating snapshot of %s: %s', volume.id, description)
-            volume.create_snapshot(description)
+            for attempt in range(5):
+                try:
+                    volume.create_snapshot(description)
```

Comment:
most of code and logic for snapshot and volume are same, I use a func wrapper to do it.